ARCHITECTURE.md

```diff
@@ -0,0 +1,194 @@
+
+We can subdivide (not break up!) our process into roughly three regions with different shapes and characteristics:
+
+- **Development Environment** roughly covers _code-to-commit_.
+- **Packaging Pipeline** roughly covers _commit-to-distribution_.
+- **Deployment and Beyond** roughly covers _distribution-to-next-rollout_.
+
+We delegate:
+
+- The **Development Environment** to a trusted project in the broader Nix Community employing community outreach to promote our cause and ensure it is at least not accidentally sabotaged.
+- The **Deployment and Beyond** by cultivating outreach and dovetailing with initiatives of, among others, the Cloud Native ecosystem.
+
+And we focus on:
+
+- The **Packaging Pipeline**
+- _Interfaces_ and _Integration_ with the other two
+
+## Architecture
+
+### Overview
+
+#### Locating Standard in the SDLC
+
+Where is Standard located in the big picture?
+
+This graphic locates Standard across the SDLC & Application Lifecycle Management (ALM).
+
+But not only that.
+It also explains how automation in itself is implemented as _code_, just as the application itself.
+Therefore, we make a distinction between:
+
+- first order application code (L1); and
+- above that, higher order supporting code as exemplified by L2 and L3.
+
+> Glossary:
+>
+> _L2 & L3_ have no clearly defined meaning.
+> They represent that we may observe multiple layers of higher order code when automating.
+> Examples could be bash scripts, configuration data, platform utility code and more.
+
+<div align="center"><img src="./artwork/sdlc.svg" width="900" /></div>
+
+#### Standard's Components and their Value Contribution
+
+What is Standard made of? And how do its components contribute value?
+
+On the left side of the graphic, we show how Standard, like an onion, is build in layers:
+
+Center to Standard is [`divnix/paisano`](https://github.com/divnix/paisano).
+This flake (i.e. "Nix library") implements two main abstractions: Block Types and Cells.
+
+_**Block Types**_ are not unlike Rust's traits or Golang's interfaces.
+They are abstract definitions of artifact classes.
+Those abstract classes implement _shared functionality_.
+
+A few examples of artifact classes in our scope are: packages, containers, scripts and manifests, among others.
+Examples of shared functionality are _push_ on containers and _build_ on packages.
+
+_**Cells**_, in turn, organize your code into related units of functionality.
+Hence, Cells are a code _orgnization principle_.
+
+On top of Paisano's abstractions, Standard implements within its scope:
+
+- a collection of Block Types; and
+- a collection of library functionality organized in Cells.
+
+On the right side of the graphic, we sketch an idea how these components are put into service for the SDLC.
+
+<div align="center"><img src="./artwork/components.svg" width="900" /></div>
+
+### Paisano (Code Organization)
+
+We already learned about Paisano's two main abstractions: Cells & Block Types.
```

Comment:
We did, but maybe a concrete code example would help in both cases somewhere earlier?

Reply:
I added some examples in `COMPARE.md`, but for an architecture, that's too early, in my view. Besides that, it's hard to capture the _concepts_ as code as opposed to a concrete _usage_ of these concepts. Here too, in an architecture overview, I don't think we should cross the boundary for concrete usage.

I know this can motivate a lot of thoughts very effectively, but I believe an architecture section also can challenge the reader's conceptual inference brain-muscle a little bit, as long as it is otherwise clear and concise.